fuel-core/src/query.rs

```diff
@@ -0,0 +1,183 @@
+use fuel_core_interfaces::{
+    common::{
+        fuel_crypto,
+        fuel_merkle,
+        fuel_types::MessageId,
+        prelude::*,
+    },
+    db::KvStoreError,
+    model::{
+        FuelBlockDb,
+        OutputProof,
+    },
+};
+
+use crate::tx_pool::TransactionStatus;
+
+#[cfg(test)]
+mod test;
```

Comment:
Why does query have its own mod with a specific test file? I would rather unit tests be attached directly to the associated functions than breaking out into a `test.rs`

Reply:
Structurally it's basically the same as if the tests were in this file, since they need to be enclosed in a test module anyways. Personally, I prefer splitting modules into smaller files to reduce the amount of scrolling when working on something and make it easier to split related code across multiple windows.

Reply:
Yeh definitely a fan of breaking into a file. Keeping the tests inline only really works with very small tests.